waspc/waspls/test/Wasp/LSP/CompletionTest.hs

```diff
@@ -0,0 +1,145 @@
+module Wasp.LSP.CompletionTest where
+
+import Control.Lens ((^.))
+import Control.Monad.Log.Pure (runLog)
+import Control.Monad.State.Strict (evalStateT)
+import qualified Data.ByteString.Lazy as BS
+import qualified Data.ByteString.Lazy.Char8 as BSC
+import Data.Foldable (find)
+import Data.List (isPrefixOf)
+import Data.Maybe (mapMaybe)
+import qualified Language.LSP.Types as LSP
+import qualified Language.LSP.Types.Lens as LSP
+import System.FilePath (replaceExtension, takeBaseName)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Golden (findByExtension, goldenVsStringDiff)
+import Wasp.Analyzer.Parser.ConcreteParser (parseCST)
+import qualified Wasp.Analyzer.Parser.Lexer as Lexer
+import Wasp.LSP.Completion (getCompletionsAtPosition)
+import Wasp.LSP.ServerState (ServerState (ServerState, _cst, _currentWaspSource, _latestDiagnostics))
+
+-- | Run test cases in ./completionTests directory
+--
+-- See 'readCompletionTest' for the format of a test case.
+test_CompletionLists :: IO TestTree
+test_CompletionLists = do
+  inputFiles <- findByExtension [".wasp"] "./waspls/test/Wasp/LSP/completionTests"
+  return $
+    testGroup "Wasp.LSP.Completion.getCompletionsAtPosition" $
+      makeCompletionTest <$> inputFiles
+
+-- | Takes a completion test case and produces the list of completion items
+runCompletionTest :: String -> String
+runCompletionTest source =
+  let (code, position) = readCompletionTest source
+      tokens = Lexer.lex code
+      parsedCST = snd $ parseCST tokens
+      serverState =
+        ServerState
+          { _currentWaspSource = code,
+            _latestDiagnostics = [],
+            _cst = Just parsedCST
+          }
+      (completionItems, _log) = runLog $ evalStateT (getCompletionsAtPosition position) serverState
+      fmtedCompletionItems = map fmtCompletionItem completionItems
+
+      fmtCompletionItem :: LSP.CompletionItem -> String
+      fmtCompletionItem item =
+        concat
+          [ "  label={",
+            show (item ^. LSP.label),
+            "} kind={",
+            show (item ^. LSP.kind),
+            "} detail={",
+            show (item ^. LSP.detail),
+            "}"
+          ]
+   in "Completion Items:\n" ++ unlines fmtedCompletionItems
+
+-- | Parses a completion test case into a pair of the wasp source code to
+-- run completion on and the position to get the completion list at.
+--
+-- = Format
+--
+-- A normal wasp file, but with two addition
+-- 1) Begins with "//! test/completion" (this is future-proofing in case we
```

Comment:
This maybe feels a tiny bit redundant at the moment, we can easily add it later if it would be helpful, and right now it doesn't help much? Although, maybe it is nice for the person checking out file by accident (for example in PR) since it indicates "hey, this is a special kind of file"?
We could have also done this by modifying the extension possibly: these files could have `.wasp.withcursor` extension for example, instead. Or `.wasp.completiontest`.
But this is also good, makes sense!